SparkPost/matchbox
# UX-702 ThemeProvider target should not be required

To reproduce the bug:
- Open up a Typescript project (Or a new one with `npx create-next-app test-project --ts`
- Try using `ThemeProvider` without the `target` prop

### What Changed
- Makes ThemeProvider target optional

### How To Test or Verify
- No easy way to verify this one since it cannot be reproduced in Libby
- Possibly running the build, copying over types from `dist` into `node_modules` in a new project

### PR Checklist
